zk_toolbox/crates/zk_inception/src/commands/prover/run.rs

```diff
@@ -0,0 +1,88 @@
+use std::{
+    path::PathBuf,
+    process::{Command, Stdio},
+};
+
+use common::{
+    cmd::{run_in_background, Cmd},
+    logger,
+};
+use config::{ChainConfig, EcosystemConfig};
+use xshell::{cmd, Shell};
+
+use super::{
+    args::run::{ProverComponent, ProverRunArgs},
+    utils::get_link_to_prover,
+};
+use crate::messages::{
+    msg_started_component, MSG_CHAIN_NOT_FOUND_ERR, MSG_MISSING_COMPONENTS_ERR,
+    MSG_RUNNING_COMPRESSOR, MSG_RUNNING_PROVER, MSG_RUNNING_PROVER_GATEWAY,
+    MSG_RUNNING_WITNESS_GENERATOR, MSG_RUNNING_WITNESS_VECTOR_GENERATOR,
+};
+
+pub(crate) async fn run(args: ProverRunArgs, shell: &Shell) -> anyhow::Result<()> {
+    let args = args.fill_values_with_prompt()?;
+    logger::debug(format!("Prover args: {:?}", args));
+    let ecosystem_config = EcosystemConfig::from_file(shell)?;
+    let chain = ecosystem_config
+        .load_chain(Some(ecosystem_config.default_chain.clone()))
+        .expect(MSG_CHAIN_NOT_FOUND_ERR);
+
+    let link_to_prover = get_link_to_prover(&ecosystem_config);
+
+    for component in args.components.expect(MSG_MISSING_COMPONENTS_ERR) {
+        match component {
+            ProverComponent::Gateway => run_gateway(&chain, &link_to_prover)?,
+            ProverComponent::WitnessGenerator => run_witness_generator(&chain, &link_to_prover)?,
+            ProverComponent::WitnessVectorGenerator => {
+                run_witness_vector_generator(&chain, &link_to_prover)?
+            }
+            ProverComponent::Prover => run_prover(&chain, &link_to_prover)?,
+            ProverComponent::Compressor => run_compressor(&chain, &link_to_prover)?,
+        }
+    }
+
```

Comment:
i don't think we have to add the ability to run them simultaneously  in one process